app/client-app/src/app/pages/Editor/components/Webhook/WebhookForm.jsx

```diff
@@ -223,156 +224,209 @@ const WebhookForm = ({ isOpen, onClose, onSubmit, webhook }) => {
           initialValues={getInitialValues(webhook)}
           onSubmit={handleFormikSubmit}
         >
-          {({ values, handleSubmit, setFieldValue }) => (
-            <Form id="myForm" onSubmit={handleSubmit}>
-              <Box p={2}>
-                <VStack align="flex-start">
-                  <TextField
-                    name="url"
-                    placeholder="Callback Url"
-                    header="Header"
-                    size="sm"
-                  />
-                  <SecretField
-                    isEditMode={isEditMode}
-                    editSecret={editSecret}
-                    handleGenerateSecret={handleGenerateSecret}
-                    values={values}
-                    setFieldValue={setFieldValue}
-                    setEditSecret={setEditSecret}
-                  />
-                  {!isEditMode && (
-                    <CreateModeSecretField
-                      values={values}
+          {({ values, handleSubmit, setFieldValue }) => {
+            useEffect(() => {
+              if (
```

Comment:
I'm thinking this might be overly complex, and is a straightforward assignment, like:

```
isChecked = webhook?.triggerCriteria?.eventTypes?.PAGE_NAVIGATION !== null;
setPageNavigationChecked(isChecked)
```
As effectively you're guarding against the state when it is `null`, so anything other than `null` is `true`. We don't need to check keys or anything, as if it has keys or not it's still always `true`.

Reply:
Good shout that works too :)